### PR TITLE
Adapter rework including breaking changes.

### PR DESCRIPTION
**Pull request by boguslaw-wojcik**
Adapter rework including breaking changes.
* Merged adapter and extended adapter to avoid further layers of complexity.
* Restructured thing and service responsibilities to better accommodate non-polling scenarios.
* Added support for connectivity and ping functionalities.

Breaking changes:
* Removed MQTT dependency from Thing Factory.
* Service is now depending on adapter for sending messages instead of MQTT instance directly.
* Thing now depends on adapter, thing state and its extendable configuration object which also includes mandatory connector service.
* Thing is now responsible for sending Inclusion, Connectivity and Ping reports.
* Introduced connector service responsible for providing connectivity and ping reports as well as connecting and disconnecting things from its data source. This is useful for non-polling scenario.

### CODE REVIEW

File: adapter/publisher.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package adapter
```
Comment (rstraczynski) [inline adapter/publisher.go:1]: Missing comments.

File: adapter/thing/battery.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+// BatteryConfig represents a thing configuration.
+type BatteryConfig struct {
+	ThingConfig   *adapter.ThingConfig
+	BatteryConfig *battery.Config
+}
+
+// NewBattery creates a thing that satisfies expectations for the battery.
+func NewBattery(
```
Comment (rstraczynski) [inline adapter/thing/battery.go:19]: Do we have a thing that contains battery service only?

**Review by rstraczynski — CHANGES_REQUESTED** [review]

File: adapter/publisher.go (hunk 1 above)
Comment (boguslaw-wojcik) [reply adapter/publisher.go:1]: Won't do this PR. Maybe next.

File: adapter/thing/battery.go (hunk 2 above)
Comment (boguslaw-wojcik) [reply adapter/thing/battery.go:19]: It's a possibility from Vinculum.

**Review by rstraczynski — APPROVED** [review]

**Pull request merged**